spring-data-mongodb/src/main/java/org/springframework/data/mongodb/core/mapping/event/ValidatingMongoEventListener.java

```diff
@@ -54,14 +55,15 @@ public ValidatingMongoEventListener(Validator validator) {
 	 */
 	@Override
 	@SuppressWarnings({ "rawtypes", "unchecked" })
-	public void onBeforeSave(Object source, DBObject dbo) {
+	public void onBeforeSave(Object obj, DBObject dbo) {
 
-		LOG.debug("Validating object: {}", source);
-		Set violations = validator.validate(source);
+		LOG.debug("Validating object: {}", obj);
+		final Object target = obj instanceof LazyLoadingProxy ? ((LazyLoadingProxy) obj).getTarget() : obj;
+		final Set violations = validator.validate(target);
 
 		if (!violations.isEmpty()) {
 
-			LOG.info("During object: {} validation violations found: {}", source, violations);
+			LOG.info("During object: {} validation violations found: {}", target, violations);
```

Comment:
unclean but didn't changed --> Logging + Throwing an Exception which may cause double logging of the same Exception / Problem.